lib/task-queue/TaskRouter.js

```diff
@@ -81,6 +87,11 @@ class TaskRouter {
 
       taskCreationLock = true;
 
+      // Normalize the url if it's not canonical
+      if (!message.data.urlIsCanonical) {
+        message.data.url = normalizeUrl(message.data.url);
```

Comment:
So we don't need `urlIsCanonical` in the messages, since we don't care if an URL is canonical or not. 😃